halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/SpinnakerEndpoints.java

```diff
@@ -54,5 +75,29 @@
     public String getBaseUrl() {
       return protocol + "://" + address + ":" + port;
     }
+
+    Service(Security security) {
+      if (security.getSsl().isEnabled()) {
+        protocol = "https";
+      }
+    }
+  }
+
+  /**
+   * Like a Service, but this has a publicly accessible endpoint (that is likely different from the one
+   * assigned within the Spinnaker cluster).
+   */
+  @Data
+  @EqualsAndHashCode(callSuper = true)
```

Comment:
Because here I do care about fields in the super-classes implementation for equality & hashing. In the other cases (subclasses of `Node`) the superclass doesn't carry useful info.